src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -679,4 +673,89 @@ private Line createLine(Network network, VoltageLevel voltageLevel1, VoltageLeve
     private Mono<Void> assertLineCreationInfosNotEmpty(LineCreationInfos lineCreationInfos) {
         return lineCreationInfos == null ? Mono.error(new NetworkModificationException(CREATE_LINE_ERROR, "Missing required attributes to create the line")) : Mono.empty();
     }
+
+    public Flux<EquipmenModificationInfos> createTwoWindingsTransformer(UUID networkUuid, UUID groupUuid, TwoWindingsTransformerCreationInfos twoWindingsTransformerCreationInfos) {
+        return assertTwoWindingsTransformerCreationInfosNotEmpty(twoWindingsTransformerCreationInfos).thenMany(
+                getNetwork(networkUuid).flatMapIterable(network -> {
+                    NetworkStoreListener listener = NetworkStoreListener.create(network, networkUuid, groupUuid, modificationRepository, equipmentInfosService);
+                    ReporterModel reporter = new ReporterModel("NetworkModification", "Network modification");
+                    Reporter subReporter = reporter.createSubReporter("TwoWindingsTransformerCreation", "Two windings transformer creation");
+
+                    return doAction(listener, () -> {
+                        // create the 2wt in the network
+                        VoltageLevel voltageLevel1 = getVoltageLevel(network, twoWindingsTransformerCreationInfos.getVoltageLevelId1());
+                        VoltageLevel voltageLevel2 = getVoltageLevel(network, twoWindingsTransformerCreationInfos.getVoltageLevelId2());
+
+                        createTwoWindingsTransformer(network, voltageLevel1, voltageLevel2, twoWindingsTransformerCreationInfos);
+
+                        subReporter.report(Report.builder()
+                                .withKey("twoWindingsTransformerCreated")
+                                .withDefaultMessage("New two windings transformer with id=${id} created")
+                                .withValue("id", twoWindingsTransformerCreationInfos.getEquipmentId())
+                                .withSeverity(new TypedValue("TWO_WINDINGS_TRANSFORMER_CREATION_INFO", TypedValue.INFO_LOGLEVEL))
+                                .build());
+
+                        // add the 2wt creation entity to the listener
+                        listener.storeTwoWindingsTransformerCreation(twoWindingsTransformerCreationInfos);
+                    }, CREATE_TWO_WINDINGS_TRANSFORMER_ERROR, networkUuid, reporter, subReporter);
+                }));
+    }
+
+    private void createTwoWindingsTransformer(Network network, VoltageLevel voltageLevel1, VoltageLevel voltageLevel2, TwoWindingsTransformerCreationInfos twoWindingsTransformerCreationInfos) {
+        TwoWindingsTransformerAdder twoWindingsTransformerAdder;
+        Optional<Substation> optS1 = voltageLevel1.getSubstation();
+        Optional<Substation> optS2 = voltageLevel2.getSubstation();
+        Substation s1 = optS1.orElse(null);
+        Substation s2 = optS2.orElse(null);
+
+        if (s1 != null) {
```

Comment:
Substation s = s1 != null ? s1 : s2;
if (s != null) s.newTwoWindingsTransformer(...
would avoid the multiline repetition, that would otherwise mean change the two first code-branches synchronouly if a parameter is to be added some time in the furure.
It could go further by initing the twoWindingsTransformerAdder in three ways (s1/s2/network) and continue the building just once.